src/transformers/tsx.ts

```diff
@@ -15,26 +15,45 @@ export default function transformTsx(code: string, node: Node, cssRoot: Root): T
       s.overwrite(match.index, match.index + match[1].length + 2,`= \`${utilityClasses} \${${match[1]}}\``)
     }
     return {
-      code: s.toString().replace(/\\:/g, '\\\\:')
+      code: s.toString().replace(/\\/g, '\\\\')
     }
   }
 }
 
 function _buildTailwindClassList(node: Node, cssRoot: Root): string | undefined {
   const classes = _parseClasses(node)
   if (classes.length) {
-    return cssRoot?.nodes?.filter(isRule).reduce((acc: any, rule: Rule) => {
-      const match = rule.selector.replace(/\\/, '').match(/([a-zA-Z0-9-]+$|[a-zA-Z0-9-]+:[a-zA-Z0-9-]+)/)
-      if (match && classes.includes(match![0])) {
-        return rule.toString().replace(/\s+/gm, ' ') + ' \\n' + acc
+    const nodes = cssRoot?.nodes?.filter(isRule).filter(rule => {
+      const match = rule.selector.replace(/\\/g, '').match(/([a-zA-Z0-9-]+$|[a-zA-Z0-9-]+:[a-zA-Z0-9-]+)/)
+      if (match && classes.includes(match[0])) {
+        return rule
       }
-      return acc
-    }, '')
+    });
+
+    const atRules = cssRoot?.nodes?.filter(isAtRule).map(atRule => {
+      atRule.nodes = atRule.nodes?.filter(isRule).filter(rule => {
+        const match = rule.selector.replace(/\\/g, '').match(/([a-zA-Z0-9-]+$|[a-zA-Z0-9-]+:[a-zA-Z0-9-]+)/);
+        if (match && classes.includes(match[0])) {
+          return rule
+        }
+      });
+      return atRule;
+    });
+    const newRoot = root({})
+
+    if (nodes) {
+      newRoot.append(nodes);
+    }
+    if (atRules) {
+      newRoot.append(atRules);
+    }
+
+    return newRoot.toString();
   }
 }
 
 function _parseClasses(node: Node) {
-  return _parseStyleDecorator(node).concat(_parseInlineClasses(node))
+  return _parseStyleDecorator(node).concat(_parseInlineClasses(node)).filter(c => c !== "")
```

Comment:
I noticed that it was including an empty string inside the class list in certain cases, so I filtered it out here. I don't think it was causing any bugs at the moment, so we can remove the filter here.